### PR TITLE
Clean up dependency tree

### PR DESCRIPTION
**Pull request by lukasj**
* remove obsolete/wrong dependencies (ie tools.jar)
* fixed osgi imports, hopefully the most important parts
* moved coverage/jacoco to extra profile
* moved server side tests and corba/jdk14 in travis to be allowed to fail (not break the build)
* fixed the build for "oracle" components

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -28,6 +29,7 @@
         <groupId>org.eclipse.ee4j</groupId>
         <artifactId>project</artifactId>
         <version>1.0.6</version>
+        <relativePath/>
```
Comment (rfelcman) [inline pom.xml:32]: Why there is empty `<relativePath/>`?

File: pom.xml [hunk 2]
```diff
@@ -241,6 +243,279 @@
...
+                <version>${project.version}</version>
+                <type>test-jar</type>
+            </dependency>
+            <dependency>
+                <groupId>org.eclipse.persistence</groupId>
+                <artifactId>org.eclipse.persistence.nosql</artifactId>
+                <version>${project.version}</version>
+                <type>test-jar</type>
+            </dependency>
+
```
Comment (rfelcman) [inline pom.xml:518]: Better solution, than my origin with `${project.version}` everywhere.

File: foundation/org.eclipse.persistence.oracle.nosql/pom.xml [hunk 3]
```diff
@@ -71,8 +71,9 @@
         </dependency>
         <!--API dependecies-->
         <dependency>
-            <groupId>jakarta.jms</groupId>
-            <artifactId>jakarta.jms-api</artifactId>
+            <groupId>javax.jms</groupId>
+            <artifactId>javax.jms-api</artifactId>
+            <version>2.0.1</version>
```
[the remark this answers is not in the document]
Comment (lukasj) [reply foundation/org.eclipse.persistence.oracle.nosql/pom.xml:76]: AQ does not support Jakarta JMS, so we have to either remove support for AQ or find a way for both to coexist

File: jpa/org.eclipse.persistence.jpa/pom.xml [hunk 4]
```diff
@@ -75,21 +77,21 @@
...
-                </configuration>
                 <executions>
                     <execution>
                         <id>bundle-manifest</id>
                         <goals>
                             <goal>manifest</goal>
                         </goals>
+                        <configuration>
+                            <instructions>
+                                <Premain-Class>org.eclipse.persistence.internal.jpa.deployment.JavaSECMPInitializerAgent</Premain-Class>
```
[the remark this answers is not in the document]
Comment (lukasj) [reply jpa/org.eclipse.persistence.jpa/pom.xml:88]: maven jar plugin handles that, why should we do it ourselves?

File: pom.xml (hunk 1 above)
Comment (lukasj) [reply pom.xml:32]: make it clear that no check to the default (`../pom.xml`) should be done by maven

File: foundation/org.eclipse.persistence.oracle.nosql/src/it/java/org/eclipse/persistence/testing/tests/eis/aq/JMSDirectConnectTest.java [hunk 5]
```diff
@@ -16,14 +16,14 @@
...
-import jakarta.jms.TopicConnectionFactory;
-import jakarta.jms.TopicSession;
+import javax.jms.JMSException;
+import javax.jms.QueueConnection;
+import javax.jms.QueueConnectionFactory;
+import javax.jms.QueueSession;
+import javax.jms.Session;
+import javax.jms.TopicConnection;
+import javax.jms.TopicConnectionFactory;
+import javax.jms.TopicSession;
```
[the remark this answers is not in the document]
Comment (lukasj) [reply foundation/org.eclipse.persistence.oracle.nosql/src/it/java/org/eclipse/persistence/testing/tests/eis/aq/JMSDirectConnectTest.java:26]: AQ does not support Jakarta JMS, so tests must use javax.jms for now

File: pom.xml (hunk 2 above)
Comment (lukasj) [reply pom.xml:518]: this part is going to be moved to an extra pom at some point

**Pull request merged**